drivers/power/adm1177/adm1177.c

```diff
@@ -0,0 +1,294 @@
+ * @brief - Gets the voltage readback from the ADM1177s I2C ADC,if the device
+ *          is initialized right and transmits the requested value.
+ *
+ * @param dev - The device structure.
+ *
+ * @param conv_voltage - The variable through which the voltage is read from
+ * 			 the ADC.
+ *
+ * @return 0 in case of success, -1 otherwise.
+******************************************************************************/
+int adm1177_voltage_readback(struct adm1177_dev *device,
+			     uint16_t* conv_voltage)
+{
+	int ret;
+
+	uint8_t data[2];
+
+	ret = no_os_i2c_read(device->i2c_desc, data, 2, 1);
+
+	if(ret)
+		return ret;
+
+	/* First byte received contains the 8 MSBs of the voltage and is
+	   left shifted by 4, then the second byte contains the 4 LSBs of
+	   the voltage and is right shifted by 4, and when the OR
+	   operation between the two bytes happen we obtain the a 12-bit
+	   value of the voltage. */
+	*conv_voltage = (data[0] << 4) | (data[1] >> 4);
+
+	return 0;
+}
+/**************************************************************************//**
+ * @brief - Gets the current readback from the ADM1177s I2C ADC,if the device
+ *          is initialized right and transmits the requested value.
+ *
+ * @param dev - The device structure.
+ *
+ * @param conv_current - The variable through which the current is read from
+ * 			 the ADC.
+ *
+ * @return 0 in case of success, -1 otherwise.
+******************************************************************************/
+int adm1177_current_readback(struct adm1177_dev *device,
+			     uint16_t* conv_current)
+{
+	int ret;
+
+	uint8_t data[2];
+
+	ret = no_os_i2c_read(device->i2c_desc, data, 2, 1);
+
+	if(ret)
+		return ret;
+
+	/* First byte received contains the 8 MSBs of the current and is
+	   left shifted by 4, then the second byte contains the 4 LSBs of
+	   the current and is right shifted by 4, and when the OR
+	   operation between the two bytes happen we obtain the a 12-bit
+	   value of the current. */
+	*conv_current = (data[0] << 4) | (data[1] >> 4);
+
+	return 0;
+}
+/**************************************************************************//**
+ * @brief - Converts the raw sample from the Readback into mV.
+ *
+ * @param raw_sample - The raw sample of the Voltage Readback from the ADC.
+ *
+ * @param v_fullscale - 0 if V_FULLSCALE_0 is selected(VRANGE bit is 0).
+ *                      1 if V_FULLSCALE_1 is selected(VRANGE bit is 1).
+ *
+ * @return - Conversion result in uV.
+ * 	     -1 if v_fullscale is set wrong.
+******************************************************************************/
+int adm1177_voltage_conversion(uint16_t raw_sample,
+			       int v_fullscale)
+{
+	switch(v_fullscale) {
+	case 0:
+		return (((uint64_t)ADM1177_V_FULLSCALE_0 * 1000000) /
+			4096) / 1000 * raw_sample;
+	case 1:
+		return (((uint64_t)ADM1177_V_FULLSCALE_0 * 1000000) /
+			4096) / 1000 * raw_sample;
+	default:
+		return -1;
```

Comment:
avoid -1, pick a meaningful error code from errno.h like -EINVAL in this case

Reply:
Will do.